src/firmware/main/world/dribbler.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <stdint.h>
+
+/**
+ * This struct represents a robot dribbler from the perspective of firmware
+ */
+struct Dribbler;
+typedef struct Dribbler Dribbler;
```

Comment:
Does 'typedef struct Dribbler' not work on its own here?

Reply:
nope, I don't think that means anything? If you think of `typedef` as taking two arguments, `struct Dribbler` is one arg, and `Dribbler` is the other. (ie the mapping is `struct Dribbler` -> `Dribbler`).

Reply:
Hmm,
 typedef struct Dribbler {} Dribbler; 
Would works when tou define it for sure.

Reply:
Hmm,
 typedef struct Dribbler {} Dribbler; 
Would works when tou define it for sure.

Reply:
So just to sanity check, I tried this, I get
```
firmware/main/app/world/dribbler.c:5:8: error: redefinition of 'Dribbler'
struct Dribbler
       ^
```
Because `typedef struct Dribbler {} Dribbler` defines the `Dribbler` struct as empty, but then we redefine it in `dribbler.c` (because it's not an empty struct, we just only want things in `dribbler.c` to be able to see what's inside. Does that make sense?

Reply:
Oh yeah, I was just saying that you can definitely define a typedef Dribbler without having to declare the struct first.

In the header we should be able to do:
typedef struct Dribbler Dribbler;

And then in the .c:
struct Dribbler {
//stuff
};

Reply:
OH I see. Nice, removed all the `struct ____` declarations in the headers.